File: airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/transform_models/UpdateStreamAttributePrimaryKeyTransform.java
/*
 * Copyright (c) 2020-2024 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.protocol.transform_models;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

/**
 * Represents the update of a stream attribute.
 */
@Getter
@AllArgsConstructor
@EqualsAndHashCode
@ToString
public class UpdateStreamAttributePrimaryKeyTransform {

  private final List<List<String>> oldPrimaryKey;
  private final List<List<String>> newPrimaryKey;

}

File: airbyte-commons-protocol/src/main/kotlin/io/airbyte/commons/protocol/transformmodels/UpdateStreamAttributePrimaryKeyTransform.kt
/*
 * Copyright (c) 2020-2026 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.commons.protocol.transformmodels

data class UpdateStreamAttributePrimaryKeyTransform(
  val oldPrimaryKey: List<List<String>>,
  val newPrimaryKey: List<List<String>>,
)
